diff -Naur src/BoltzTraP.F90 patch/BoltzTraP.F90
--- src/BoltzTraP.F90	2013-11-20 16:29:01.000000000 +0100
+++ patch/BoltzTraP.F90	2018-11-09 16:35:09.668181699 +0100
@@ -76,7 +76,7 @@
   integer :: un_phdosxyz = 54
 
 
-  WRITE(6,*) '================ BoltzTraP vs 1.2.5 ============='
+  WRITE(6,*) '================ BoltzTraP vs 1.2.5 - coeff_file_patch ============='
   CALL gtfnam(deffn,errfn,iproc)
   tauinput=.FALSE.
   readvvin=.FALSE.
@@ -226,7 +226,7 @@
        CALL fite4(cell, bs)
        write(6,*) 'Subroutine fite4 executed'
 
-       call out_engre(un_engre, egap, bs)
+       call out_engre(un_engre, egap, bs, cell)
 
      
      END IF
@@ -246,7 +246,7 @@
        lifetbs%icut2 = bs%icut2
        ALLOCATE(lifetbs%engre(lifetbs%nwave,bs%icut1:bs%icut2))
        CALL fite4(cell, lifetbs)
-       call out_engre(un_lfengre, egap, lifetbs)
+       call out_engre(un_lfengre, egap, lifetbs, cell)
      ENDIF
 
      IF(bandstyle == "GENE_PHON" .or. bandstyle == 'PHON') THEN
@@ -258,7 +258,7 @@
        phonbs%icut2 = phonbs%nband
        ALLOCATE(phonbs%engre(phonbs%nwave,phonbs%icut1:phonbs%icut2))
        CALL fite4(cell, phonbs)
-       call out_engre(un_phengre, phegap, phonbs)
+       call out_engre(un_phengre, phegap, phonbs, cell)
      END IF
   ELSE
 ! already calculated: read in lattice points and engre from files.
@@ -310,7 +310,7 @@
          projbs%icut2 = bs%icut2
          ALLOCATE(projbs%engre(projbs%nwave,projbs%icut1:projbs%icut2))
          CALL fite4(cell, projbs)
-         call out_engre(un_projengre, egap, projbs)
+         call out_engre(un_projengre, egap, projbs, cell)
          CALL dos(cell,bs,spinorbit,egap,icut12,icut22,lifetbs%engre,tauinput, &
 &             un_dos, un_dosxy, un_dosxyz, un_v2dos, &
 &             projbs%engre, un_dosproj(iproj))
diff -Naur src/m_bandstructure.F90 patch/m_bandstructure.F90
--- src/m_bandstructure.F90	2013-11-20 15:21:57.000000000 +0100
+++ patch/m_bandstructure.F90	2018-11-09 16:36:02.649414707 +0100
@@ -101,23 +101,32 @@
     if(associated(bs%latt_points)) deallocate (bs%latt_points)
   END SUBROUTINE destroy_bandstructure
 
-  subroutine out_engre(unitgre, egap, bs)
+  subroutine out_engre(unitgre, egap, bs, cell)
     implicit none
     integer, intent(in) :: unitgre
     type(bs_type), intent(inout) :: bs
     real(8), intent(in) :: egap
+    type(cell_type), intent(in) :: cell
   
     integer :: i
 
     rewind (unitgre)
-    WRITE(unitgre) egap 
+    WRITE(unitgre) egap
     WRITE(unitgre) bs%nwave
+    
+     WRITE(123,*) egap, bs%nwave, cell%nsym, size(cell%symop(1,1,:))
+     WRITE(123,'(9f25.14)') MATMUL(cell%aac_dir,cell%p2c_dir)
+     WRITE(123,'(2000f5.1)') cell%symop
+!     
     DO i=1,bs%nwave
       WRITE(unitgre) bs%latt_points(1:3,i)
+       WRITE(123,*) bs%latt_points(1:3,i)
     ENDDO
     WRITE(unitgre) bs%icut1,bs%icut2
+    WRITE(123,*) bs%icut1,bs%icut2
     DO i=bs%icut1,bs%icut2
       WRITE(unitgre) bs%engre(1:bs%nwave,i)
+      WRITE(123,'(50000e25.14)') bs%engre(1:bs%nwave,i)
     ENDDO
   end subroutine out_engre
   
